pkg/azuredisk/azuredisk_v2.go

```diff
@@ -405,11 +406,16 @@ func (d *DriverV2) StartControllersAndDieOnExit(ctx context.Context) {
 		klog.Errorf("Failed to initialize NodeAvailabilityController. Error: %v. Exiting application...", err)
 		os.Exit(1)
 	}
-	err = controller.WatchObject(mgr, source.Kind{Type: &v1.Namespace{}})
+	err = controller.CacheNamespace(mgr)
 	if err != nil {
 		klog.Errorf("Failed to watch Namespace. Error: %v. Exiting application...", err)
 		os.Exit(1)
 	}
+	err = controller.CacheCRD(mgr, sharedState)
```

Comment:
If we are going with the predicate approach here, this is no longer caching, this method should be something like `WatchForDriverUninstall`

Reply:
Sure! When we finalize which approach to go, I'll change it.